src/cpp/preprocessor/aie2/aie2_blob_preprocessor_input.h

```diff
@@ -269,5 +270,69 @@ class aie2_asm_preprocessor_input : public aie2_blob_transaction_preprocessor_in
     aie2_blob_transaction_preprocessor_input::set_args(mc_code, patch_json, control_packet, libs, libpaths, ctrlpkt);
   }
 };
+
+class config_preprocessor_input : public aie2_blob_transaction_preprocessor_input
+{
+  uint32_t col = 0;
+protected:
+  void readconfigjson(std::istream& patch_json);
+  void add_pdi(const boost::property_tree::ptree& pinstance);
+  void add_instance(const boost::property_tree::ptree& pinstance);
+  class argument {
+    public:
+    std::string name;
+    std::string type;
+    std::string offset;
+
+    argument(const std::string& na, const std::string& ty, const std::string& off): name(na), type(ty), offset(off) {}
+    argument(const argument& rhs) = default;
+    argument& operator=(const argument& rhs) = default;
+    argument(argument &&s) = default;
+    ~argument() = default;
+  };
+
+  struct function {
+    std::string name;
+    std::vector<argument> arguments;
+  };
+
+  std::string mangle_function_name(const function& func) {
```

Comment:
what are we using in _xbreplay_ for function name mangling? Can we reuse the same code as it gone through the scrutiny?